### PR TITLE
[Azure Analysis Services] Add gateway info to version 0714 and version 0801

### PR DESCRIPTION
**Pull request by taiwu**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by taiwu** [comment]
Sync and add older version as requested in
https://github.com/Azure/azure-rest-api-specs/pull/1518

**Comment by taiwu** [comment]
I am not a vendor.

**Comment by olydis** [comment]
@taiwu Please complete the checklist. Regarding vendor message: Seems like the PR was still tagged as "cla-not-required", so you can ignore that. (classification is flaky sometimes)

**Comment by taiwu** [comment]
@olydis thanks for your reply, could you please take a look at the change?

**Comment by sergey-shandar** [comment]
@ravbhatnagar New API version for Analysis Services.

**Comment by ravbhatnagar** [comment]
@sergey-shandar @taiwu  This is difficult to review. Can you please make a commit using the contents from the existing API-version. And then add the changes? Is it possible to do it now?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 1]
```diff
@@ -607,6 +653,60 @@
         }
       }
     },
+    "GatewayDetails": {
+      "description": "The gateway details.",
+      "type": "object",
+      "properties": {
+        "gatewayResourceId": {
```
Comment (ravbhatnagar) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Is gateway resource exposed through ARM?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 2]
```diff
@@ -381,6 +381,48 @@
       }
     }
   },
+  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AnalysisServices/servers/{serverName}/listGatewayStatus": {
```
Comment (ravbhatnagar) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Why do you need a  separate POST API to get the gateway status? Cant it be obtained by GET /servers/{servername}?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 3]
```diff
@@ -607,6 +653,60 @@
...
+          "readOnly": true,
+          "description": "Uri of the DMTS cluster. Only supported with api versions 2017-07-14 or higher."
+        }
+      }
+    },
+    "GatewayListStatusLive": {
+      "description": "Status of gateway is live",
+      "type": "object",
+      "properties": {
+        "status": {
```
Comment (ravbhatnagar) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: What are the valid values for the status property? Can this be an enum?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 4]
```diff
@@ -381,6 +381,48 @@
...
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": { "$ref": "#/definitions/GatewayListStatusLive" }
+          },
+          "400": {
+            "description": "Gateway not live or not configured.",
+            "schema": { "$ref": "#/definitions/GatewayListStatusError" }
+          },
+          "404": {
```
Comment (ravbhatnagar) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: There could be other error codes that this API returns. So its best to capture the error conditions by defining a "default" response and including the error object definition in that.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

**Comment by ravbhatnagar** [comment]
@sergey-shandar  - SDK can start reviewing.

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json (hunk 2 above)
Comment (taiwu) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: By design, the spec can be found at section 3.9 of http://aka.ms/asarm/
Get /servers/{servername} only contains the gateway configuration info.

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json (hunk 4 above)
Comment (taiwu) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: sure, will add default.

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json (hunk 1 above)
Comment (taiwu) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Yes

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 5]
```diff
@@ -381,6 +381,48 @@
...
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": { "$ref": "#/definitions/GatewayListStatusLive" }
+          },
+          "400": {
```
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: If 4xx responses are supposed to throw an exception in generated code (it looks like that was the intention), they can unfortunately not be modeled like that but have to go into a `default` response. See https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/creating-swagger.md#negative-responses for details. I'd recommend documenting status codes and their corresponding description in the description of the default response.

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 6]
```diff
@@ -607,6 +653,60 @@
         }
       }
     },
+    "GatewayDetails": {
+      "description": "The gateway details.",
+      "type": "object",
+      "properties": {
+        "gatewayResourceId": {
+          "type": "string",
+          "description": "Gateway resource to be associated with the server. Only supported with api versions 2017-07-14 or higher."
```
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Not sure whether it makes sense to document "Only supported with api versions 2017-07-14 or higher." - this very Swagger is tied to an API version, so it's kind of redundant information. On the other hand, you may have thought about what people read in docs generated from this Swagger (and I'm not sure how explicit API versions are called out in our auto-gened docs). But generally, I haven't seen this pattern before! @ravbhatnagar Guidance?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json (hunk 3 above)
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: +1

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 7]
```diff
@@ -607,6 +653,60 @@
...
+      "description": "Status of gateway is error",
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/GatewayErrors",
+          "description": "Error of the list gateway status."
+        }
+      }
+    },
+    "GatewayErrors": {
```
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Why is this called `GatewayErrors`? It's used in a property called `error` and also by itself looks like it represents a single error.

File: specification/analysisservices/resource-manager/readme.md [hunk 8]
```diff
@@ -39,6 +39,24 @@ input-file:
 - Microsoft.AnalysisServices/2016-05-16/analysisservices.json
```
Comment (olydis) [inline specification/analysisservices/resource-manager/readme.md]: Please flip order (most recent first). This has no influence on behavior, but makes it consistent with the other RP's configuration in this repo 🙂

**Review by olydis — CHANGES_REQUESTED** [review]
see comments

**Comment by taiwu** [comment]
Fixed, please take a look.
Small adjustment, the v714 was a copy of v801, which is now based on v516.

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 9]
```diff
@@ -607,6 +649,63 @@
...
+    "GatewayListStatusLive": {
+      "description": "Status of gateway is live",
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string",
+          "enum": [
+            "Live"
+          ],
+          "description": "Live message of list gateway."
```
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: add `x-ms-enum` part (set model as string to `true`)

**Review by olydis — CHANGES_REQUESTED** [review]
Minor comment, but please take a look at the validation tool outputs: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/263248184#L665
Is there anything you can resolve? Note that the `x-ms-enum` comment is one of the issues flagged by the validator, please review these issues.

**Comment by taiwu** [comment]
Updated, the rest are all examples and operation list.
The latest build failure is not related to my latest commit.
@olydis could you please take a look?

File: specification/analysisservices/resource-manager/readme.md [hunk 10]
```diff
@@ -26,10 +26,28 @@ These are the global settings for the AnalysisServices API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2016-05
+tag: package-2017-07
```
Comment (michael-falk) [inline specification/analysisservices/resource-manager/readme.md]: Could you change package-2017-07 to package-2017-08-beta so it refers to latest and #1542 can be closed.

File: specification/analysisservices/resource-manager/readme.md [hunk 11]
```diff
@@ -26,7 +26,7 @@ These are the global settings for the AnalysisServices API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2017-07
+tag: package-2017-07-08-beta
```
Comment (michael-falk) [inline specification/analysisservices/resource-manager/readme.md]: tag: package-2017-07-08-beta?
do you mean:
tag: package-2017-08-beta

**Review by olydis — APPROVED** [review]
LGTM, descriptions of model types can be made consistently end with `.`

**Comment by olydis** [comment]
@ravbhatnagar All your concerns addressed?

File: specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json [hunk 12]
```diff
@@ -379,6 +379,44 @@
...
+          },
+          {
+            "name": "serverName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z][a-z0-9]*$",
+            "minLength": 3,
+            "maxLength": 63,
+            "description": "The name of the Analysis Services server. It must be at least 3 characters in length, and no more than 63."
```
Comment (olydis) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Please remove ` It must be at least 3 characters in length, and no more than 63.`, don't restate constraints expressed in Swagger by other means.

Comment (taiwu) [inline specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json]: Sure, will do, but I just copy paste from previous one....
Can you just merge?

**Review by ravbhatnagar — APPROVED** [review]

**Pull request merged**